test/test-queue-foreach-delete.c

```diff
@@ -69,7 +69,7 @@ static const unsigned first_handle_number_fs_event = 0;
   static uv_##name##_t (name)[3];                                             \
   static unsigned name##_cb_calls[3];                                         \
                                                                               \
-  static void name##2_cb(uv_##name##_t* handle) {                             \
+  static void name##2_cb(uv_##name##_t* handle, ...) {                        \
```

Comment:
I don't think this is the right fix. With some ABIs the calling convention of variadic functions isn't compatible with that of non-variadic functions, like when the normal calling convention stipulates that the callee cleans the stack.

What you could do is change the macro like this:
```c
#define DEFINE_GLOBALS_AND_CBS(name, ...)
```
Then change this line (and ditto the other callbacks) to:
```c
static void name##2_cb(__VA_ARGS__) {
```
Then change the uses to:
```c
DEFINE_GLOBALS_AND_CBS(idle, uv_idle_t* handle)
// ...
DEFINE_GLOBALS_AND_CBS(fs_event, uv_fs_event_t* handle, int status)
```
It's kind of clunky. It could be more DRY if we could use `__VA_OPT__(,)` but unfortunately that's non-standard in C89/C99.

On the upside, with these changes you no longer need the casts on line 96-98.

Reply:
Thank you for your feedback!
Are you hereby granting your OK to use a C99 feature (__VA_ARGS__)?
I thought libuv allowed only C89 to come in while going through the contributing guide
```
* Use C89-compliant syntax. In other words, variables can only be declared at
  the top of a scope (function, if/for/while-block).
```
By the way, I have always wondered why libuv needs to be compiled against GNU89 rather than C89 standard
https://github.com/libuv/libuv/blob/5fd480678c230b9bf57298d4a9d761c5d2dbd0c7/configure.ac#L32
Could you shed light upon this?

Reply:
The C89 restriction is true for code in src/ and include/. For stuff in test/ a subset of C99 is acceptable. The rule of thumb for src/ and include/ is that it needs to work with MSVC 2008 because of Python 2.

I think the `-std=gnu89` switch is because we use `inline` in one or two places.

Reply:
I have got them changed as per your request.